### PR TITLE
feat(DTFS2-6182): replace get deal and get deals graphql endpoints with REST endpoints

### PR DESCRIPTION
**Pull request by ChristianMcc**
## Introduction
We need to standardise the TFM API by moving away from having both gql and REST endpoints. We will move the gql endpoints to REST.

## Resolution
Replace the get deal and get deals graphql endpoints with REST endpoints as part of the gql -> rest migration in TFM API.
I have deprecated the deals query and only kept the deals light query as the former was unused within DTFS.

### CODE REVIEW

File: trade-finance-manager-api/src/v1/routes.js [hunk 1]
```diff
@@ -85,6 +86,10 @@ openRouter.route('/deals/submit').put(dealSubmit.submitDealPUT);
 
 authRouter.route('/deals/submitDealAfterUkefIds').put(dealSubmit.submitDealAfterUkefIdsPUT);
 
+authRouter.route('/deals').get(dealController.getDeals);
+
+authRouter.route('/deals/:dealId').get(dealController.getDeal);
```
Comment (francescastocco) [inline trade-finance-manager-api/src/v1/routes.js:91]: should you add validation for deal ID here?
`....get(validation.dealIdValidation, handleValidationResult, dealController.getDeal);`

**Review by francescastocco — CHANGES_REQUESTED** [review]
Looks good :)

File: trade-finance-manager-api/src/v1/routes.js (hunk 1 above)
Comment (ChristianMcc) [reply trade-finance-manager-api/src/v1/routes.js:91]: Great catch! Definitely should thanks for spotting

**Review by francescastocco — APPROVED** [review]

**Review by AlexBramhill — COMMENTED** [review]
reviewed non-mappings files

File: trade-finance-manager-api/src/v1/graphql-mappings/filters/filterActivities.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  ALL: 'all-activity',
+  COMMENT: 'comments-only',
+};
+
+const filter = (activities, FILTER_VALUE) => activities.filter((activity) => activity.type === FILTER_VALUE);
+
+const filterByComment = (activities) => filter(activities, 'COMMENT');
+
+const filterActivities = (activities, filtersObj) => {
+  if (!filtersObj?.filterType || filtersObj?.filterType === FILTER_TYPE.ALL) {
```
[the remark this answers is not in the document]
Comment (AlexBramhill) [reply trade-finance-manager-api/src/v1/graphql-mappings/filters/filterActivities.js:11]: will rereview once merged into main ticket

File: trade-finance-manager-ui/server/api.js [hunk 3]
```diff
@@ -59,34 +67,39 @@ const getFacilities = async (queryParams) => {
...
+      headers: generateHeaders(token),
+      data: queryParams,
+    });
+    if (response.data) {
+      return {
+        deals: response.data.deals,
+        count: response.data.count,
+      };
+    }
+    return response.data;
```
[the remark this answers is not in the document]
Comment (AlexBramhill) [reply trade-finance-manager-ui/server/api.js]: added todo

**Pull request merged**